Makefile

```
@@ -144,7 +144,7 @@ test-deleted-renamed-referenced-files:
 
 .PHONY: test-soft_failure-no-reference
 test-soft_failure-no-reference:
-	@! git --no-pager grep -E -e 'record_soft_failure\>.*\;' --and --not -e '([a-zA-Z]+#[a-zA-Z-]*[0-9]+|fate.suse.com/[0-9]+|\$reference)' lib/ tests/
+	@! git --no-pager grep -E -e 'record_soft_failure\>.*\;' --and --not -e '([a-zA-Z]+#[a-zA-Z-]*[0-9]+|fate.suse.com/[0-9]+|\$$(reference|bsc))' lib/ tests/
```

```suggestion
	@! git --no-pager grep -E -e 'record_soft_failure\>.*\;' --and --not -e '([a-zA-Z]+#[a-zA-Z-]*[0-9]+|fate.suse.com/[0-9]+|\$$(reference|bsc|bugref))' lib/ tests/
```

Looking at the rest of the changes, this will likely end up being like the above code as a transitional state, and later unify everything with bugref, in the future far far away

---

IMO it could also recognize a marker like the python `# noqa`  one

---

If there's a valid `record_soft_failure` without any `bsc#`/`$bsc`/etc. it could be annotated to pass CI